source/Ship.cpp

```diff
@@ -3946,12 +3946,39 @@ double Ship::BestFuel(const string &type, const string &subtype, double defaultF
 		double jumpRange = baseAttributes.Get("jump range");
 		if(!jumpRange)
 			jumpRange = System::DEFAULT_NEIGHBOR_DISTANCE;
+
+		double fuel = baseAttributes.Get("jump fuel");
+
+		double mass = Mass();
+		double driveMassExp = 0;
+		double driveMassRef = 400;
+		double driveDistExp = 0;
+		double driveDistRef = 100;
+
+		if((baseAttributes.Get("drive mass reference") || baseAttributes.Get("drive mass exponent")) && mass > 0)
```

Comment:
```suggestion
		if((baseAttributes.Get("drive mass reference") || baseAttributes.Get("drive mass exponent")))
```
Mass cannot be negative.